pkg/recipes/driver/gitconfig.go

```diff
@@ -88,19 +100,30 @@ func addSecretsToGitConfig(secrets v20231001preview.SecretStoresClientListSecret
 	return nil
 }
 
-// Unset the git credentials information from .gitconfig by running
-// git config --global --unset url<template_path_domain_with_credentails>.insteadOf
-func unsetSecretsFromGitConfig(secrets v20231001preview.SecretStoresClientListSecretsResponse, templatePath string) error {
-	urlConfigKey, _, err := getURLConfigKeyValue(secrets, templatePath)
+// setGitConfigForDir sets a conditional include directive in the global Git configuration file.
+// This function modifies the global Git configuration to include a specific Git configuration file
+// when the repository is located in the given working directory. The `includeIf` directive is used
+// to conditionally include the configuration file located at "<workingDirectory>/.git/config".
+func setGitConfigForDir(workingDirectory string) error {
+	cmd := exec.Command("git", "config", "--global", fmt.Sprintf("includeIf.gitdir:%s/.path", workingDirectory), workingDirectory+"/.git/config")
+	_, err := cmd.Output()
 	if err != nil {
-		return err
+		return errors.New("failed to add conditional include directive")
```

Comment:
Don't we need to see the details of the error? Why can't we wrap the error into the other one?